main/src/main/scala/sbt/Plugins.scala

```diff
@@ -345,4 +347,13 @@ ${listConflicts(conflicting)}""")
     }
     hasGetterOpt getOrElse false
   }
+
+  /** Debugging method to time how long it takes to run various compilation tasks. */
+  private[this] def timed[T](label: String, log: Logger)(t: => T): T = {
+    val start = System.nanoTime
+    val result = t
+    val elapsed = System.nanoTime - start
+    log.debug(label + " took " + (elapsed / 1e6) + " ms")
+    result
+  }
```

Comment:
This brings to 5 the number of `timed` functions. Maybe that we should just define it `private[sbt]` somewhere